pocketutils/misc/klgists.py

```diff
@@ -0,0 +1,829 @@
+	"""Same as zip(), but raises an IndexError if the lengths don't match."""
+	iters = [iter(axis) for axis in args]
+	n_elements = 0
+	failures = []
+	while len(failures) == 0:
+		n_elements += 1
+		values = []
+		failures = []
+		for axis, iterator in enumerate(iters):
+			try:
+				values.append(next(iterator))
+			except StopIteration:
+				failures.append(axis)
+		if len(failures) == 0:
+			yield tuple(values)
+	if len(failures) == 1:
+		raise IndexError("Too few elements ({}) along axis {}".format(n_elements, failures[0]))
+	elif len(failures) < len(iters):
+		raise IndexError("Too few elements ({}) along axes {}".format(n_elements, failures))
+
+
+def only(sequence: Iterable[Any]) -> Any:
+	"""
+	Returns either the SINGLE (ONLY) UNIQUE ITEM in the sequence or raises an exception.
+	Each item must have __hash__ defined on it.
+	:param sequence: A list of any items (untyped)
+	:return: The first item the sequence.
+	:raises: ValarLookupError If the sequence is empty
+	:raises: MultipleMatchesError If there is more than one unique item.
+	"""
+	st = set(sequence)
+	if len(st) > 1:
+		raise MultipleMatchesError("More then 1 item in {}".format(sequence))
+	if len(st) == 0:
+		raise LookupFailedError("Empty sequence")
+	return next(iter(st))
+
+
+def read_lines_file(path: str, ignore_comments: bool = False) -> Sequence[str]:
+	"""
+	Returns a list of lines in a file, potentially ignoring comments.
+	:param path: Read the file at this local path
+	:param ignore_comments: Ignore lines beginning with #, excluding whitespace
+	:return: The lines, with surrounding whitespace stripped
+	"""
+	lines = []
+	with open(path) as f:
+		line = f.readline().strip()
+		if not ignore_comments or not line.startswith('#'):
+			lines.append(line)
+	return lines
+
+
+def read_properties_file(path: str) -> Mapping[str, str]:
+	"""
+	Reads a .properties file, which is a list of lines with key=value pairs (with an equals sign).
+	Lines beginning with # are ignored.
+	Each line must contain exactly 1 equals sign.
+	:param path: Read the file at this local path
+	:return: A dict mapping keys to values, both with surrounding whitespace stripped
+	"""
+	lines = read_lines_file(path, ignore_comments=False)
+	dct = {}
+	for i, line in enumerate(lines):
+		if line.startswith('#'): continue
+		if line.count('=') != 1:
+			raise ParsingError("Bad line {} in {}".format(i+1, path))
+		parts = line.split('=')
+		dct[parts[0].strip()] = parts[1].strip()
+	return dct
+
+
+class Comparable:
+	"""A class that's comparable. Just implement __lt__. Credit ot Alex Martelli on https://stackoverflow.com/questions/1061283/lt-instead-of-cmp"""
+
+	def __eq__(self, other):
+		return not self < other and not other < self
+
+	def __ne__(self, other):
+		return self < other or other < self
+
+	def __gt__(self, other):
+		return other < self
+
+	def __ge__(self, other):
+		return not self < other
+
+	def __le__(self, other):
+		return not other < self
+
+
+def json_serial(obj):
```

Comment:
I use [orjson](https://github.com/ijl/orjson) now, but I'd leave this in `klgists.py` if it works.